usr/share/rear/restore/COVE/default/400_restore_with_cove.sh

```diff
@@ -0,0 +1,370 @@
+        { COVE_INSTALLATION_TOKEN="$(UserInput -I COVE_INSTALLATION_TOKEN -C -r -t 0 -p "$prompt")" ; } 2>>/dev/$SECRET_OUTPUT_DEV
+    fi
+
+    {
+        local new_installer_path="${COVE_TMPDIR}/cove#v1#${COVE_INSTALLATION_TOKEN}#.run"
+        if [ "$COVE_INSTALLER_PATH" != "$new_installer_path" ]; then
+            # Rename the Backup Manager installer to enable Unified Installation Flow
+            mv "$COVE_INSTALLER_PATH" "$new_installer_path"
+            COVE_INSTALLER_PATH="$new_installer_path"
+        fi
+
+        [ -x "$COVE_INSTALLER_PATH" ] || chmod +x "$COVE_INSTALLER_PATH"
+    } 2>>/dev/$SECRET_OUTPUT_DEV
+
+    local target_install_dir="${TARGET_FS_ROOT}/${COVE_REAL_INSTALL_DIR#/}"
+
+    # First, try OverlayFS to redirect the installation to a disk
+    if ! cove_try_overlayfs "$COVE_INSTALL_DIR" "$target_install_dir"; then
+        # Create symlinks to redirect the installation to a disk
+        if [ "$COVE_INSTALL_DIR" != "$COVE_REAL_INSTALL_DIR" ]; then
+            local target="$target_install_dir"
+            local link_name="$COVE_INSTALL_DIR"
+            cove_create_symlink "${target}" "${link_name}" || return $?
+        else
+            cove_dirs=(bin etc lib sbin share temp var/log var/storage)
+            for cove_dir in "${cove_dirs[@]}"; do
+                local target="${target_install_dir}/${cove_dir}"
+                local link_name="${COVE_INSTALL_DIR}/${cove_dir}"
+                cove_create_symlink "$target" "$link_name" || return $?
+            done
+        fi
+    fi
+
+    UserOutput ""
+    UserOutput "Installing Backup Manager..."
+    "$COVE_INSTALLER_PATH" --target "${COVE_TMPDIR}/mxb" 1>&7 2>&8 || return $?
+
+    # Extract the ReaR tarball because the installer does not do it in the rescue environment
+    mkdir -p "${target_install_dir}/rear"
+    tar -xf "${COVE_TMPDIR}/mxb/rear.tar.gz" --strip-components=1 -C "${target_install_dir}/rear"
+
+    # Try to copy site.conf if it does not exist at the target.
+    # It happens in case of symlinks used to redirect the installation.
+    [ -e "${target_install_dir}/rear/etc/rear/site.conf" ] \
+        || cp "${COVE_INSTALL_DIR}/rear/etc/rear/site.conf" "${target_install_dir}/rear/etc/rear/site.conf" \
+        || true
+}
+
+# Stops ProcessController process
+function cove_stop_pc() {
+    local pid
+    pid="$(ps aux | awk -v pc_name=ProcessController '$0 ~ pc_name && !/awk/ {print $2}')"
+    [ -z "$pid" ] || { /bin/kill -TERM "$pid" && \
+    while [ -n "$pid" ]; do \
+        sleep 1; \
+        pid="$(get_pc_pid)"; \
+    done }
+}
+
+# Print the welcome message
+UserOutput "
+The System is now ready for restore. The Backup Manager installer will be
+downloaded and run automatically. If any required parameters have not been
+provided, you will be prompted to enter them."
+
+# Read parameters from boot options. The existing values can be overridden
+# by values passed via boot options.
+read -r cmdline </proc/cmdline
```

Comment:
I'd do this in an earlier stage, e.g. `verify` and not here in `rescue`. The reason is to fail early and without touching the disks if we believe that something is missing.